### ISSUE TITLE
Windows 11 box not reporting HyperV support correctly

### ISSUE THREAD

**Issue by olix3001** [hashicorp/vagrant#13313]
## Problem
I've been trying to create box with `generic-x64/fedora39` image, but even though it is marked as having hyperv support I get the following error:
```
The box you're attempting to add doesn't support the provider
you requested. Please find an alternate box or use an alternate
provider. Double-check your requested provider to verify you didn't
simply misspell it.

If you're adding a box from HashiCorp's Vagrant Cloud, make sure the box is
released.

Name: generic-x64/fedora39
Address: https://vagrantcloud.com/api/v2/vagrant/generic-x64/fedora39
Requested provider: hyperv (x64)
```

I thought It might be something wrong with the box itself, but when changed to `generic/fedora37` (version 4.3.8) it got even stranger showing the following error:
```
The box you're attempting to add has no available version that
matches the constraints you requested. Please double-check your
settings. Also verify that if you specified version constraints,
that the provider you wish to use is available for these constraints.

Box: generic/fedora37
Address: https://vagrantcloud.com/api/v2/vagrant/generic/fedora37
Constraints: 4.3.8
Available versions: 4.2.0, 4.2.2, 4.2.4, 4.2.6, 4.2.8, 4.2.10, 4.2.12, 4.2.14, 4.2.16, 4.3.0, 4.3.2, 4.3.4, 4.3.6, 4.3.8
```
When not specifying version for the first box, i also get `Constraints: >= 0; Available versions: 4.3.8`.

It seems like the two examples above should just work as versions matching constraints are available.

## Replication
```yaml
Windows edition: Windows 11 Pro
Version: 22H2
OS build: 22621.2861
```

Vagrant is installed from the official website (version 2.4.0), HyperV is enabled, and all vcredist versions are installed (I've seen that these sometimes can cause strange issues).

### Debug output
https://gist.github.com/olix3001/c892853b00db19c5629964929b7a5c89

In the Gist above there are two files: `first.txt` and `second.txt`:
- `first.txt` contains logs from running `vagrant box add generic-x64/fedora39 --provider hyperv --debug`
- `second.txt` contains logs from running `vagrant box add generic/fedora39 --provider hyperv --box-version 4.3.8 --debug`

### Expected behavior
Box gets installed successfully and can be used in `Vagrantfile`.

### Actual behavior
Strange errors are thrown even though HyperV is supported and correct versions are available.

**Comment by mdpdesign** [comment]
I'm experiencing the same issue but with different box, I'm using box version constraint in my `Vagrantfile` - `lb.vm.box_version = "4.3.10"`

```
Bringing machine 'lb1' up with 'hyperv' provider...
==> lb1: Verifying Hyper-V is enabled...
==> lb1: Verifying Hyper-V is accessible...
==> lb1: Box 'generic/ubuntu2204' could not be found. Attempting to find and install...
    lb1: Box Provider: hyperv
    lb1: Box Version: 4.3.10
==> lb1: Loading metadata for box 'generic/ubuntu2204'
    lb1: URL: https://vagrantcloud.com/api/v2/vagrant/generic/ubuntu2204
The box you're attempting to add has no available version that
matches the constraints you requested. Please double-check your
settings. Also verify that if you specified version constraints,
that the provider you wish to use is available for these constraints.

Box: generic/ubuntu2204
Address: https://vagrantcloud.com/api/v2/vagrant/generic/ubuntu2204
Constraints: ~> 4.3.10
Available versions: 4.0.0, 4.0.2, 4.0.4, 4.1.0, 4.1.2, 4.1.4, 4.1.6, 4.1.8, 4.1.10, 4.1.12, 4.1.14, 4.1.16, 4.1.18, 4.1.20, 4.2.0, 4.2.2, 4.2.4, 4.2.6, 4.2.8, 4.2.10, 4.2.12, 4.2.14, 4.2.16, 4.3.0, 4.3.2, 4.3.4, 4.3.6, 4.3.8, 4.3.10
```

```
vagrant box add generic/ubuntu2204 --box-version 4.3.10
==> box: Loading metadata for box 'generic/ubuntu2204'
    box: URL: https://vagrantcloud.com/api/v2/vagrant/generic/ubuntu2204
The box you're attempting to add has no available version that
matches the constraints you requested. Please double-check your
settings. Also verify that if you specified version constraints,
that the provider you wish to use is available for these constraints.

Box: generic/ubuntu2204
Address: https://vagrantcloud.com/api/v2/vagrant/generic/ubuntu2204
Constraints: 4.3.10
Available versions: 4.0.0, 4.0.2, 4.0.4, 4.1.0, 4.1.2, 4.1.4, 4.1.6, 4.1.8, 4.1.10, 4.1.12, 4.1.14, 4.1.16, 4.1.18, 4.1.20, 4.2.0, 4.2.2, 4.2.4, 4.2.6, 4.2.8, 4.2.10, 4.2.12, 4.2.14, 4.2.16, 4.3.0, 4.3.2, 4.3.4, 4.3.6, 4.3.8, 4.3.10
```

Vagrant version:

```
vagrant version
Installed Version: 2.4.0
Latest Version: 2.4.0

You're running an up-to-date version of Vagrant!
```

Box URL `https://vagrantcloud.com/api/v2/vagrant/generic/ubuntu2204` lists version `4.3.10` as `active` and `hyperv` provider is available.

A workaround I was able to use was to add that box "manually":

- Get full URL of the box for Hyper-V provider
- Run `vagrant box add ...` with that URL, box is added as `v0`
- Once the box is added, rename version in `.vagrant.d/boxes/<box-name>` to desired version - in my case `4.3.10`
- Run `vagrant up` - it should work now (worked in my case)

```
vagrant box add generic/ubuntu2204 https://vagrantcloud.com/generic/boxes/ubuntu2204/versions/4.3.10/providers/hyperv/amd64/vagrant.box
==> box: Box file was not detected as metadata. Adding it directly... 
==> box: Adding box 'generic/ubuntu2204' (v0) for provider:
box: Downloading: https://vagrantcloud.com/generic/boxes/ubuntu2204/versions/4.3.10/providers/hyperv/amd64/vagrant.box
box:
==> box: Successfully added box 'generic/ubuntu2204' (v0) for ''!
```

**Comment by chrisroberts** [comment]
Hi there,

This issue is a duplicate of #13274 which has been fixed and will be included in the 2.4.1 release which is currently building now.

Cheers!
